zapr.go

```diff
@@ -187,5 +234,37 @@ func NewLogger(l *zap.Logger) logr.Logger {
 	return logr.New(newLoggerWithExtraSkip(l, 1))
 }
 
+// NewLogger creates a new logr.Logger using the given Zap Logger to log
+// and applies additional options.
+func NewLoggerWithOptions(l *zap.Logger, opts ...Option) logr.Logger {
+	zl := newLoggerWithExtraSkip(l, 1)
+	zl.logErrorFieldName = "error"
+	for _, option := range opts {
+		option(zl)
+	}
+	return logr.New(zl)
+}
+
+// Option is one additional parameter for NewLoggerWithOptions.
+type Option func(*zapLogger)
+
+// WithLogNumeric controls whether a numeric log level is added to
+// each log message. The empty string disables this, a non-empty
+// string is the key for the additional field. Errors use -1 and
+// panics -2.
+func WithLogNumeric(key string) Option {
+	return func(zl *zapLogger) {
+		zl.logNumericFieldName = key
```

Comment:
This is wordy for not much value.  I'd streamline the name like `numericLevelKey` or `levelKey`